languages/cpp/src/BitwardenClient.cpp

```diff
@@ -0,0 +1,127 @@
+#include "BitwardenClient.h"
+#include <iostream>
+#include <string>
+
+BitwardenClient::BitwardenClient(ClientSettings clientSettings) : library(nullptr), commandRunner(nullptr), isClientOpen(false), projects(nullptr), secrets(nullptr) {
```

Comment:
Similar to the other language bindings, we don't want to publicly expose `ClientSettings` since the consumers should only be able to configure `apiUrl` and `identityUrl`. `deviceType` should be left to null, and `userAgent` should be hard coded to `Bitwarden CPP-SDK`

Reply:
@Hinton I provided `identityUrl` and `apiUrl ` to be configurable with `BitwardenSettings` object. If empty parameters are sent then default values for these two are used. Please check if that is what you want.